src/panel/visual-regression.test.ts

```diff
@@ -69,26 +69,17 @@ describe.each(parallelize(fixtures))("%s", (id, { rendererUrl }) => {
     });
   };
 
-  beforeEach(async () => {
-    await page.goto(rendererUrl, { waitUntil: "load" });
-    await page.mouse.move(0, 0);
-    await delay(500);
-  });
-
-  describe("landscape viewport", () => {
-    beforeEach(async () => {
+  describe("Go to page, test landscape and portrait viewport", () => {
+    it("matches landscape and portrait snapshot", async () => {
```

Comment:
I think it could be nice still having these split into separate `landscape` and `portrait` tests with a `beforeAll` or something.

Reply:
I would prefer the same, however I kept running into an issue using `beforeAll` where the page would close after the first run - then all the remaining tests would fail.

Reply:
This is the error I get when using `beforeAll`

```
    Protocol error (Page.navigate): Session closed. Most likely the page has been closed.

      72 |   describe("Go to page, test landscape and portrait viewport", () => {
      73 |     beforeAll(async () => {
    > 74 |       await page.goto(rendererUrl, { waitUntil: "load" });
         |                  ^
      75 |       await page.mouse.move(0, 0);
      76 |       await delay(500);
      77 |     });

```

Reply:
Ah - that's happening because of the execution order, https://jestjs.io/docs/en/setup-teardown.html. `beforeAll` getting triggered before the `beforeEach` `page.close()`

Reply:
A (crude) fix would be to force the navigation to run after the `page.close()`

```js
      let hasRan = false;

      beforeEach(async () => {
        if (hasRan) {
          return;
        }
        hasRan = true;
        await page.goto(rendererUrl, { waitUntil: "load" });
  
        await page.mouse.move(0, 0);
      })

  
      it("matches landscape", async () => {
        await page.setViewport({ width: 1200, height: 600 });
        await delay(500);
        await matchSnapshot({ viewport: "landscape" });
      });
      it("matches portrait", async () => {
        await page.setViewport({ width: 600, height: 1200 });
        await delay(500);
        await matchSnapshot({ viewport: "portrait" });
      });
    })

Reply:
Running with `if (page.url() != rendererUrl)` was surprisingly ~30% slower than the above (or by just rendering the render in the landscape test)

Reply:
Ah okay cool. I'll go with your suggestion. Shame that the more idiomatic `beforeAll` fails due the somewhat obscure execution order. 

Although I now see the `afterEach` in `setupTests.ts` that closes the page - so now it makes sense.

Reply:
On reflection - it looks like the page is setup and torn down in `setupTests` but then the page is only ever used in the visual regression?

Perhaps we should move page setup and teardown to visual-regression?

Reply:
> Perhaps we should move page setup and teardown to visual-regression?

We did have that in the past but from what I remember,  using `jest-puppeteer` was solving some determinism issues we were having during tests.

Totally open to having the setup configured in this test alone if we get a performance benefit! Going to ask some Formidafolks today about getting something like [Percy](https://percy.io/) setup